### PR TITLE
Add support for generating interface configuration in /etc/network/interfaces for multiple management interfaces

### PR DESCRIPTION
**Pull request by geogchen**
Signed-off-by: George Chen gechen@microsoft.com

#### Why I did it
The interfaces template, interfaces.j2, currently hardcodes to eth0 and therefore cannot support multiple management interfaces.  This pull request introduces change to support multiple management interfaces.

#### How I did it
1) Update interfaces.j2 to iterate over MGMT_INTERFACE where it is currently not, and refer to interface name rather than hardcoded eth0.
2) Update minigraph.py to process mgmt_intf in sorted order.  This is needed because in Python 3.x, the dictionary keys are presented in such a way that eth1 in [t0-sample-graph-two-mgmt.xml](https://github.com/Azure/sonic-buildimage/compare/master...geogchen:dev/gechen/multi_mgmt_interface?expand=1#diff-0a80119cc01db12d4686c311fc7be9281147ff487247d611610f4784e526f454) appears first, and it gets rewritten to be eth0 into MGMT_INTERFACE.  To prevent the interface-reordering, process the interfaces in sorted order.

#### How to verify it
Unit test.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

#### Description for the changelog
Add support to generating multiple management interfaces for /etc/network/interfaces 

#### A picture of a cute animal (not mandatory but encouraged)

### CODE REVIEW

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -1432,7 +1432,8 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
     results['MGMT_INTERFACE'] = {}
     mgmt_intf_count = 0
     mgmt_alias_reverse_mapping = {}
-    for key in mgmt_intf:
+    sorted_keys = sorted(mgmt_intf.keys(), key=lambda x : x[0].lower())
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py:1435]: >sorted [](http://example.com/codeflow?start=18&length=6)

Is `sorted` really needed?

Comment (geogchen) [reply src/sonic-config-engine/minigraph.py:1435]: without sorting, in Py 3 eth1 is returned first, and will be re-written as eth0 in [this line](https://github.com/Azure/sonic-buildimage/pull/11204/files/6351b9a9a5b5bb0daa508842128ce131cd3b4ec5#diff-466ae5de99730b4f3dde9c932e9086536e3a0226858c735fbf776453e8d86fafR1441).

Comment (qiluo-msft) [reply src/sonic-config-engine/minigraph.py:1435]: Strictly speaking, better to use natural sort, like Management1~Management11

**Review by qiluo-msft — APPROVED** [review]

File: src/sonic-config-engine/minigraph.py (hunk 1 above)
Comment (geogchen) [reply src/sonic-config-engine/minigraph.py]: changed to use natsort

**Pull request merged**